internal/controller/mimiralertmanagerconfig/mimiralertmanagerconfig_controller.go

```diff
@@ -70,15 +77,41 @@ func (r *MimirAlertManagerConfigReconciler) Reconcile(ctx context.Context, req c
 		}
 	}
 
-	return ctrl.Result{}, r.handleCreationAndChanges(ctx, amc)
+	return ctrl.Result{}, r.handleCreationAndChanges(ctx, amc, mc)
+}
+
+func (r *MimirAlertManagerConfigReconciler) createMimirClient(ctx context.Context, amc *domain.MimirAlertManagerConfig) (*mimirapi.MimirClient, error) {
+
+	auth, err := utils.ExtractAuth(ctx, r.Client, amc.Spec.Auth, amc.ObjectMeta.Namespace)
+	if err != nil {
+		return nil, fmt.Errorf("failed to extract authentication settings: %w", err)
+	}
+	// if no auth provided init empty object to avoid error
+	if auth == nil {
```

Comment:
shouldn't it be the responsibility of the called function (extractauth) to do that, this remove duplicate code in the mimirrule controller too